lib/response.js

```diff
@@ -26,156 +26,13 @@ function Response(body, opts) {
 	this.status = opts.status;
 	this.statusText = http.STATUS_CODES[this.status];
 	this.headers = new Headers(opts.headers);
-	this.body = body;
-	this.bodyUsed = false;
-	this.size = opts.size;
 	this.ok = this.status >= 200 && this.status < 300;
-	this.timeout = opts.timeout;
-
-}
-
-/**
- * Decode response as json
- *
- * @return  Promise
- */
-Response.prototype.json = function() {
 
-	return this._decode().then(function(text) {
-		return JSON.parse(text);
-	});
+	Body.call(this, body, opts);
```

Comment:
That's only for the default. The whole system is designed in which the module passes down its own `Promise` to the modules it calls.